### ISSUE TITLE
points layer + 3D + perspective projection results in white image on MacOS

### ISSUE THREAD

**Issue by maweigert** [napari/napari#3077]
## 🐛 Bug

Rendering a points layer in 3D and changing the projection from orthogonal to perspective results in a completely white image on MacOS

## To Reproduce

```python
import numpy as np
import napari 

viewer = napari.Viewer()
viewer.add_points(np.random.uniform(0,100,(1000,3)))
viewer.dims.ndisplay = 3
viewer.camera.perspective=1e-10 # or any other positive value
```

## Expected behavior

A rendered points layer (as when setting `viewer.camera.perspective=0` 

## Environment

```
napari: 0.4.11.dev61+g3bd58bde.d20210722
Platform: macOS-10.15.7-x86_64-i386-64bit
System: MacOS 10.15.7
Python: 3.8.8 (default, Feb 24 2021, 13:46:16) [Clang 10.0.0 ]
Qt: 5.15.2
PyQt5: 5.15.4
NumPy: 1.19.5
SciPy: 1.7.0
Dask: 2021.04.1
VisPy: 0.6.6

OpenGL:
- GL version: 2.1 ATI-3.10.18
- MAX_TEXTURE_SIZE: 16384

Screens:
- screen 1: resolution 1792x1120, scale 2.0
```

**Comment by jni** [comment]
Yup, I can reproduce, thanks for the example @maweigert! It's also reversible. To me it suggests a vispy bug, we'll have to drill down and make a vispy-only reproducer though.

**Comment by jni** [comment]
Ok, so I was able to *sort of* reproduce in VisPy with this code:

```python
import sys
import numpy as np
from vispy import scene
from vispy.scene import visuals

pos = np.random.normal(size=(1000, 3), scale=0.2)
canvas = scene.SceneCanvas(keys='interactive', show=True, bgcolor='w')
view = canvas.central_widget.add_view()
view.camera = 'arcball'

scatter = visuals.Markers(parent=view.scene)
scatter.antialias = 0
scatter.set_data(
    pos=pos,
    edge_color=(0.0, 0.0, 0.0, 1.0),
    face_color=(0.6, 0.5, 0.4, 1.0),
    size=.03,
    scaling=True,
)

if __name__ == '__main__' and sys.flags.interactive == 0:
    canvas.app.run()
```

roughly adapted from https://github.com/vispy/vispy/issues/1166

Here's the result with me moving through the perspective values with shift+right click. You can see that just above 0 (isometric), the markers become enormous and fill the field of view, but otherwise things are ok:

https://user-images.githubusercontent.com/492549/126742530-cdc4986e-2b05-42be-83ae-4cb9ada211bb.mov

But on your example, I can never change the perspective enough to make a difference. The *big* difference is that `size=.03`, which is indeed tiny. Adding that to the napari code is enough to make the visualisation "rescuable".

At any rate, even when you can rescue it, the way that perspective interacts with point sizes is suuuuper weird. (ie what's happening in your example @maweigert is that the points become absolutely ginormous.) I'm gonna hand off to @sofroniewn now as he has played much more with marker sizes. Maybe @alisterburt has ideas also. 😬

**Comment by tlambert03** [comment]
yes, this has been a known vispy bug since the [original addition](https://github.com/napari/napari/pull/2636#issuecomment-830988629) of the slider.  (see also https://github.com/napari/napari/issues/2637)

**Comment by sofroniewn** [comment]
> I'm gonna hand off to @sofroniewn now as he has played much more with marker sizes.

Might be something good for Owen when he joins @jni

**Comment by brisvag** [comment]
Closed by #3494.
